lib/util.js

```diff
@@ -781,7 +781,8 @@ var util = {
         });
         self.runTo();
       });
-    }
+    };
+    constructor._PromiseDependency = PromiseDependency;
```

Comment:
It seems messy to attach the Promise constructor to the Request constructor and then to reference that outside of the request class (in managed upload). 

This might be a good time to refactor the `util.setPromisesDependency` method to store the Promise constructor, and then add a `util.getPromisesDependency` method to retrieve it.